include/syncmer_hash.hpp

```diff
@@ -0,0 +1,135 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2021, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2021, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Hossein Eizadi Moghadam <hosseinem AT fu-berlin.de>
+ * \brief Provides syncmer_hash.
+ */
+
+#pragma once
+
+#include <seqan3/alphabet/views/complement.hpp>
+#include <seqan3/core/detail/strong_type.hpp>
+#include <seqan3/search/views/kmer_hash.hpp>
+#include <seqan3/search/views/minimiser_hash.hpp>
+#include <seqan3/utility/views/zip.hpp>
+#include "syncmer.hpp"
+#include "shared.hpp"
+
+namespace seqan3::detail
+{
+//!\brief seqan3::views::syncmer_hash's range adaptor object type (non-closure).
+//!\ingroup search_views
+struct syncmer_hash_fn
+{
+    /*!\brief Store the window_size and the subwindow_size and return a range adaptor closure object.
+    * \param[in] window_size       The number of values in one window
+    * \param[in] subwindow_size    The number of values to be checked as subwindow.
```

Comment:
I think, for the syncmer_hash functions we can now use k-mers and s-mers as we are now working directly on the sequences.